packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1506,9 +1506,9 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     }
 
     // Notify when there is a new pending index/value. Also, if there is a pending value, it needs to send undefined.
-    if (newPendingIndex !== undefined || newPendingValue || this._hasPendingValue) {
+    if (newPendingIndex !== undefined || newPendingValue !== undefined || this._hasPendingValue) {
       onPendingValueChanged(
-        newPendingIndex ? currentOptions[newPendingIndex] : undefined,
+        currentOptions[newPendingIndex!],
```

Comment:
Is the `!` really necessary? Hasn't it been guarded out in line 1509?

Reply:
Unfortunately no. Good catch. Because of the `||` condition, it doesn't necessarily have a value :( Now it should be ok.